gdal/ogr/ogrsf_frmts/sqlite/ogrsqliteutility.h

```diff
@@ -36,26 +36,32 @@
 
 #include <set>
 #include <string>
+#include <memory>
 
-typedef struct
+class SQLResult
 {
-    char** papszResult;
-    int nRowCount;
-    int nColCount;
-    char *pszErrMsg;
-    int rc;
-} SQLResult;
+    public:
+        SQLResult(char** papszResult, int nRow, int nCol);
+        ~SQLResult ();
+
+        int         RowCount() const;
+        int         ColCount() const;
+        void        LimitRowCount(int nLimit);
+
+        const char* GetValue(int iColumnNum, int iRowNum) const;
+        int         GetValueAsInteger(int iColNum, int iRowNum) const;
+    private:
```

Comment:
While I was at it, I made the class more encapsulated and also removed the error message member variable since those are really only used in `SQLQuery`. So now,  a `SQLResult` will always hold the result of a successful query.